src/diffusers/models/unets/unet_2d.py

```diff
@@ -240,6 +242,10 @@ def __init__(
         self.conv_act = nn.SiLU()
         self.conv_out = nn.Conv2d(block_out_channels[0], out_channels, kernel_size=3, padding=1)
 
+    def _set_gradient_checkpointing(self, module, value=False):
+        if hasattr(module, "gradient_checkpointing"):
+            module.gradient_checkpointing = value
```

Comment:
I notice we use `hasattr(module, "gradient_checkpointing")` in around half the cases of `_set_gradient_checkpointing` and module names in the other half, no need to change it for now but we could look at making it uniform at some point.